packages/contracts-bedrock/src/L1/SuperchainConfig.sol

```diff
@@ -129,51 +136,76 @@ contract SuperchainConfig is Initializable, ISemver {
 
     /// @notice Getter for the systemOwner address.
     function systemOwner() public view returns (address systemOwner_) {
-        systemOwner_ = Slot.getAddress(SYSTEM_OWNER_SLOT);
+        systemOwner_ = Storage.getAddress(SYSTEM_OWNER_SLOT);
     }
 
     /// @notice Getter for the initiator address.
     function initiator() public view returns (address initiator_) {
-        initiator_ = Slot.getAddress(INITIATOR_SLOT);
+        initiator_ = Storage.getAddress(INITIATOR_SLOT);
     }
 
     /// @notice Getter for the vetoer address.
     function vetoer() public view returns (address vetoer_) {
-        vetoer_ = Slot.getAddress(VETOER_SLOT);
+        vetoer_ = Storage.getAddress(VETOER_SLOT);
     }
 
     /// @notice Getter for the guardian address.
     function guardian() public view returns (address guardian_) {
-        guardian_ = Slot.getAddress(GUARDIAN_SLOT);
+        guardian_ = Storage.getAddress(GUARDIAN_SLOT);
     }
 
     /// @notice Getter for the delay time.
     function delay() public view returns (uint256 delay_) {
-        delay_ = Slot.getUint(DELAY_SLOT);
+        delay_ = Storage.getUint(DELAY_SLOT);
     }
 
     /// @notice Getter for the maxPause time.
     function maxPause() public view returns (uint256 maxPause_) {
-        maxPause_ = Slot.getUint(MAX_PAUSE_SLOT);
+        maxPause_ = Storage.getUint(MAX_PAUSE_SLOT);
     }
 
-    /// @notice Getter for the paused address.
+    /// @notice Getter for the paused value.
     function paused() public view returns (bool paused_) {
-        paused_ = Slot.getUint(PAUSED_TIME_SLOT) > block.timestamp;
+        paused_ = Storage.getUint(PAUSED_TIME_SLOT) > block.timestamp;
+    }
+
+    /// @notice Getter for the paused address.
+    function pausedUntil() public view returns (uint256 paused_) {
+        paused_ = Storage.getUint(PAUSED_TIME_SLOT);
     }
 
     /// @notice Pauses withdrawals.
```

Comment:
I am curious how you feel about reducing the ABI and having branching logic in `pause` based on whether or not it is already paused. The big difference between the branches is sloading the pause end time vs using the block timestamp. This will reduce some duplicate code and make the administration of pausing more simple. The difference here is a signed tx for extend pause can be invalidated by the pause duration running out but it will become valid again if the system is paused again, assuming no nonce invalidations

Reply:
Nice idea. The fact that auth and the storage written to is identical on both seems like a good indication that they can be combined. It simplifies both the interface and operations so sounds good.